### ISSUE TITLE
how should we "orchestrate" cleanup

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/reproman#467]
TL;DR question(s):  
- should I just `reproman run ... datalad drop` to lighten up the remote copy of the dataset?  
- how to remove entire dataset -- smth like `p="$PWD" && cd .. && datalad remove --nocheck "$p"`?

My thoughts out loud were:

If I schedule execution of job(s) on a remote service (one of the datalad orchestrators in my case), data will be obtained/transferred, job executed, results fetched, and input/output data will be left residing on the remote host.  Leaving data behind for the next steps makes total sense, but I wonder now how from the client (where `reproman run` is executed) we could take care about cleaning up after ourselves:  we might want to 
- drop all (or just output) data
- remove dataset entirely

use case: given a collection of datasets I would like to schedule (via submitters) running mriqc and fmriprep whenever a subject data arrives. So on the client box I will develop a script which 
- will monitor for new subject directories appearing (across a collection of datasets set in a hierarchy) or just not having yet a corresponding preprocessing done
- it will establish desired output subdatasets (if not there yet) and  `reproman run` (without `--follow`) preprocessing (using containers) steps (there is no dependencies yet between the steps, but they might want the same inputs, so they might collide there... yet to see how datalad would puke) on available HPC
- it will query previously ran jobs via `reproman jobs`, and would trigger fetching of the results
- whenever all preprocessing pieces for a given subject are complete, it should trigger clean up on remote side for that subject (again via `reproman run` but probably with `local` submitter, and with `--follow`)

Does such workflow sound reasonable, or could be better @kyleam ? ;)

**Comment by kyleam** [comment]
> TL;DR question(s):
> 
> * should I just `reproman run ... datalad drop` to lighten up the remote copy of the dataset?

I'd think that should work.  As mentioned on the call, it will create commits with the .reproman/ metadata changes.  I personally wouldn't like having those commits in my logs, but it sounds like you wouldn't mind.

Perhaps we'd want to introduce a flag to run that says "don't save the metadata directory if there are no other changes".

> * how to remove entire dataset -- smth like `p="$PWD" && cd .. && datalad remove --nocheck "$p"`?

Hmm, I'm guess that one would work.

> Leaving data behind for the next steps makes total sense, but I wonder now how from the client (where `reproman run` is executed) we could take care about cleaning up after ourselves: we might want to
> 
> * drop all (or just output) data
> * remove dataset entirely

Right, I think both of those are something reproman should support, but at the moment it doesn't.  Along the same lines, the run-based orchestrators currently leave tar.gz files lying around that we should clean up:

https://github.com/ReproNim/reproman/blob/903b7aa8b31deca8455ddfbb5adff14d65e7bf14/reproman/support/jobs/orchestrators.py#L975-L979
